scripts/log2changes.pl

```diff
@@ -37,6 +37,15 @@ sub nicedate {
     return $date;
 }
 
+sub printmsg {
+    my ($p, $msg)=@_;
+    while(length($msg) > 77) {
+        print $p.substr($msg, 0, 77, "")."\n";
+        $p="  ";
```

Comment:
To avoid outputting lines with spaces-only (untested):
```perl
$o=substr($msg, 0, 77, "");
if $o ne "" {
   $o=$p.$o
}
print $o."\n";
```

Fixed earlier in fde1cb9247792dfeeb6e44a07c84ed3957cb4707.

Reply:
I can't see them actually occurring? I tried this:
~~~diff
diff --git a/scripts/log2changes.pl b/scripts/log2changes.pl
index f4c06b750..fbaf177a9 100755
--- a/scripts/log2changes.pl
+++ b/scripts/log2changes.pl
@@ -41,11 +41,12 @@ sub printmsg {
     my ($p, $msg)=@_;
     while(length($msg) > 77) {
         print $p.substr($msg, 0, 77, "")."\n";
         $p="  ";
     }
-    print "$p$msg\n";
+    print "$p$msg\n" if($msg !~ /^\s+\z/);
+    print "HERE\n" if($msg =~ /^\s+\z/);
 }
 
~~~
... but I don't see any `HERE` output when I try this?

Reply:
I'm not sure why the test doesn't trigger, but the spaces are definitely there in the output:
<img width="315" alt="Screen Shot 2022-11-14 at 11 31 24" src="https://user-images.githubusercontent.com/1446897/201638399-81f46409-0948-413d-8b20-a10e642b8c43.png">

Also, my patch was incomplete, here's one that fixes the above (and multiple issues with my patch):
```diff
--- log2changes-a.pl	2022-11-14 10:23:38 +0000
+++ log2changes-b.pl	2022-11-14 10:27:51 +0000
@@ -40,9 +40,16 @@
 sub printmsg {
     my ($p, $msg)=@_;
     while(length($msg) > 77) {
-        print $p.substr($msg, 0, 77, "")."\n";
+        $o=substr($msg, 0, 77, "");
+        if($o eq "") {
+            $p="";
+        }
+        print $o."\n";
         $p="  ";
     }
+    if($msg eq "") {
+        $p="";
+    }
     print "$p$msg\n";
 }
```

Reply:
Ah, now I get it. I think it can be simplified to:
~~~difff
index f4c06b750..ee8a753eb 100755
--- a/scripts/log2changes.pl
+++ b/scripts/log2changes.pl
@@ -41,10 +41,13 @@ sub printmsg {
     my ($p, $msg)=@_;
     while(length($msg) > 77) {
         print $p.substr($msg, 0, 77, "")."\n";
         $p="  ";
     }
+    if($msg eq "") {
+        $p = "";
+    }
     print "$p$msg\n";
 }
 
 print
~~~
... since it it isn't likely that lines longer than 77 columns would return a "" (or even spaces-only) for those initial 77 columns.